Fix LimitedLibraryPicker not popping up.

kurt-anderson-prod opened the pull request:
# Summary

If you have multiple ViewControllers nested, the prompt for the LimitedLibraryPicker will not prompt on the right viewcontroller, which makes nothing actually happen.

Added a quick cycler to go through the tree until it finds the top most view controller.

## Test Plan

I have run this on many different instances to make sure the code is still solid. So far I have found no issues with this small code snippet. 

### What's required for testing (prerequisites)?

None required. Just have an app with react-native-permissions installed. 

### What are the steps to reproduce (after prerequisites)?

Go multiple viewcontrollers down within an app and try to call the LimitedLibraryPicker. On 3.0.0 this does not work. On the current version of the app is does. 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |
| Android |    -iOS Only Bug-   |

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [ X] I have tested this on a device and a simulator
- [ X] I added the documentation in `README.md`
- [X ] I mentioned this change in `CHANGELOG.md`
- [ X] I updated the typed files (TS and Flow)
- [ X] I added a sample use of the API in the example project (`example/App.js`)

This is a very simple commit. The above are not necassary. Not much at all has changed.

kurt-anderson-prod committed:
Fix Prompt Not Happening on Nested ViewControllers

If you have multiple ViewControllers nested, the prompt for the LimitedLibraryPicker will not prompt on the right viewcontroller, which makes nothing actually happen.

Added a quick cycler to go through the tree until it finds the top most view controller.

kurt-anderson-prod committed:
Update presentLimited to new TopView Controller

zoontek commented:
Thanks! I believe this is exactly the goal of `RCTPresentedViewController` (in `RCTUtils`)

https://github.com/facebook/react-native/blob/46be292f671c70aac4ecc178c96e3a2a6a3d16da/React/Base/RCTUtils.m#L549

Could you use this instead instead ?

zoontek commented on ios/PhotoLibrary/RNPermissionHandlerPhotoLibrary.m:
```diff
@@ -56,9 +56,17 @@ - (void)openLimitedPhotoLibraryPickerWithResolver:(RCTPromiseResolveBlock _Nonnu
 
     UIViewController* rootViewController = [[UIApplication sharedApplication].keyWindow rootViewController];
-    [[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:rootViewController];
+    [[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:topViewController];
```
Add the import:

```objc
#import <React/RCTUtils.h>
```

Replace all of this with:

```objc
UIViewController *topViewController = RCTPresentedViewController();
[[PHPhotoLibrary sharedPhotoLibrary] presentLimitedLibraryPickerFromViewController:topViewController];
```

kurt-anderson-prod replied:
I agree, that would definitely be the better way to do it. Tested and it works!

kurt-anderson-prod committed:
Make Code More Efficient

As pointed out by Zoontek, there is a function to find the top view controller in React Native. Importing that function, and then updated the topViewController to use that function.

kurt-anderson-prod commented:
Changes pushed up!

zoontek commented:
@princekurt Perfect, thank you!

zoontek commented:
I released a new beta with this change 👍
